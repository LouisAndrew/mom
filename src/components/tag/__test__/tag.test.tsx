import React from 'react';
import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';

import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import Tag from '..';

describe('Tag component', () => {
    const mockContent = 'A tag!';
    const mockHandleClick = jest.fn(() => {});

    afterEach(cleanup);

    const el: React.ReactElement = (
        <Tag content={mockContent} handleClick={mockHandleClick} />
    );

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('should render with text content correctly', () => {
        const { queryByText } = render(el);
        const tag: HTMLElement | null = queryByText(mockContent);

        if (tag) {
            expect(true).toBe(true);
        } else fail();
    });

    it('should react to user click correctly', () => {
        const { getByText } = render(el);
        const tag: HTMLElement = getByText(mockContent);

        fireEvent.click(tag);

        expect(mockHandleClick).toBeCalled();
    });

    // it('matches snapshot', () => {

    //     const tree = renderer.create(el).toJSON()
    //     expect(tree).toMatchSnapshot()
    // })
});
