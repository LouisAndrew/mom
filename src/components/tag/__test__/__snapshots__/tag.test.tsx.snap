// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component matches snapshot 1`] = `
<span
  className="sc-AxheI buEslG"
  color="dark.3"
  fontFamily="body"
  fontSize={
    Array [
      1,
      1,
      2,
      2,
    ]
  }
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [
          Class {
            "_dispatchInstances": null,
            "_dispatchListeners": null,
            "_targetInst": null,
            "altKey": null,
            "bubbles": null,
            "button": null,
            "buttons": null,
            "cancelable": null,
            "clientX": null,
            "clientY": null,
            "ctrlKey": null,
            "currentTarget": [Function],
            "defaultPrevented": null,
            "detail": null,
            "dispatchConfig": null,
            "eventPhase": null,
            "getModifierState": [Function],
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": null,
            "metaKey": null,
            "movementX": [Function],
            "movementY": [Function],
            "nativeEvent": null,
            "pageX": null,
            "pageY": null,
            "relatedTarget": [Function],
            "screenX": null,
            "screenY": null,
            "shiftKey": null,
            "target": null,
            "timeStamp": [Function],
            "type": null,
            "view": null,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onKeyDown={
    [MockFunction] {
      "calls": Array [
        Array [
          Class {
            "_dispatchInstances": null,
            "_dispatchListeners": null,
            "_targetInst": null,
            "altKey": null,
            "bubbles": null,
            "button": null,
            "buttons": null,
            "cancelable": null,
            "clientX": null,
            "clientY": null,
            "ctrlKey": null,
            "currentTarget": [Function],
            "defaultPrevented": null,
            "detail": null,
            "dispatchConfig": null,
            "eventPhase": null,
            "getModifierState": [Function],
            "isDefaultPrevented": [Function],
            "isPropagationStopped": [Function],
            "isTrusted": null,
            "metaKey": null,
            "movementX": [Function],
            "movementY": [Function],
            "nativeEvent": null,
            "pageX": null,
            "pageY": null,
            "relatedTarget": [Function],
            "screenX": null,
            "screenY": null,
            "shiftKey": null,
            "target": null,
            "timeStamp": [Function],
            "type": null,
            "view": null,
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  role="button"
  tabIndex={1}
>
  A tag!
</span>
`;
